src/client/resources/web/js/message_parsing.js

```diff
@@ -40,19 +78,23 @@ function isValidColor(color) {
     return /^#[0-9a-fA-F]{6}$/.test(color); // Allow valid hex colors (e.g., #FF0000)
 } 
 
-// HTML escaping including backticks and backslashes to prevent template literal and escape sequence exploits
+/**
+ * HTML escaping including backticks and backslashes to prevent template literal and escape sequence exploits.
+ * @param {string} unsafe
+ * @returns {string}
+ */
 function escapeHtml(unsafe) {
     if (typeof unsafe !== 'string') {

```

Comment:
Maybe you'll want to remove this runtime check?

Reply:
No, just want to prevent shenanigans from users. Being a bit defensive in these situations seems good to me.

Reply:
I'll change the type of `unsafe` to `unknown` then.